pkg/controller/account/ec2.go

```diff
@@ -42,22 +47,35 @@ func (r *ReconcileAccount) InitializeSupportedRegions(reqLogger logr.Logger, acc
 	customerTags := r.getCustomTags(reqLogger, account)
 
 	// Create go routines to initialize regions in parallel
-
 	for _, region := range regions {
 		go r.InitializeRegion(reqLogger, account, region.Name, regionAMIs[region.Name], vCPUQuota, ec2Notifications, ec2Errors, creds, managedTags, customerTags) //nolint:errcheck // Unable to do anything with the returned error
 	}
 
+	var regionInitFailedRegion []string
+	regionInitFailed := false
 	// Wait for all go routines to send a message or error to notify that the region initialization has finished
 	for i := 0; i < len(regions); i++ {
 		select {
 		case msg := <-ec2Notifications:
 			reqLogger.Info(msg)
 		case errMsg := <-ec2Errors:
-			reqLogger.Error(errors.New(errMsg), errMsg)
+			regionInitFailed = true
+			// If we fail to initialize the desired region we want to fail the account
+			reqLogger.Error(errors.New(errMsg.ErrorMsg), errMsg.ErrorMsg)
+			regionInitFailedRegion = append(regionInitFailedRegion, errMsg.Region)
 		}
 	}
-
-	reqLogger.Info("Completed initializing all supported regions")
+	// // If an account is BYOC or CCS and region initialization fails we want to fail the account else output success log
+	if regionInitFailed && len(regions) == 1 {
```

Comment:
I'm curious here why the `== 1` check? Should we not be checking `len(regionInitFailedRegion) > 0`

Reply:
We only care if region init failed for accounts that are only initializing a single region, as that means that the future install will also fail, but it will be a longer time before it fails.  The only way an account would be initializing a single region would be if it's CCS or ROSA, essentially if it's not being created out of the accountpool and being created by an accountclaim.

In the case of non-ccs accounts, we're initializing on all regions that are available, and those _may_ fail because of the way that AWS makes your "first attempt to spend money in each region fail" for whatever business reason they had for that decision.  Since that's expected behavior we don't want to put the account into an error state for that reason alone.

The argument _could_ be made that we should just say "is not a non-ccs account" instead of `len(regions)` but I'm not sure if that would be any more clear.

This question being asked though means that the comment on line 68 isn't descriptive enough to describe why we're checking for this, and probably should be updated.